spanner/value.go

```diff
@@ -5592,12 +5592,26 @@ func encodeProtoEnumArray(len int, at func(int) reflect.Value) (*proto3.Value, e
 	return listProto(vs...), nil
 }
 
+// spannerTag contains metadata about a struct field's spanner tag.
+type spannerTag struct {
+	// Generated is true if the field is a generated column.
+	Generated bool
+}
+
 func spannerTagParser(t reflect.StructTag) (name string, keep bool, other interface{}, err error) {
 	if s := t.Get("spanner"); s != "" {
 		if s == "-" {
 			return "", false, nil, nil
 		}
-		return s, true, nil, nil
+		parts := strings.Split(s, ",")
+		name = parts[0]
+		tag := spannerTag{}
+		for _, part := range parts[1:] {
```

Comment:
Why do we skip the first result here? Is it because it should always be parsed as the column name?

Can we add some documentation (code comments) that explains what can go into this tag, and how it should be used (including this new tag)?

Reply:
>Why do we skip the first result here? Is it because it should always be parsed as the column name?

Yes, and if we don't skip it, it will be a breaking change for existing users.

Reply:
Added doc.go, now instead of comma we optional tags will be separate by a semicolan (;) and `->` is supported similar to GORM https://gorm.io/docs/models.html#Fields-Tags